### PR TITLE
Make controller in DBMobileSharedApplication and DBDesktopSharedApplication a weak reference

### PR DESCRIPTION
**Pull request by YufeiG**
Addresses https://github.com/dropbox/dropbox-sdk-obj-c/issues/165

These are singletons so they were retaining the controller forever.
This may require a major update as it is changing behavior that isn't backwards compatible. However I can't see this breaking anything for others because the controller should be held strongly in the view hierarchy.

### CODE REVIEW

File: Source/ObjectiveDropboxOfficial/Platform/ObjectiveDropboxOfficial_iOS/DBOAuthMobile-iOS.m [hunk 1]
```diff
@@ -15,7 +15,7 @@
 
 @implementation DBMobileSharedApplication {
   UIApplication *_sharedApplication;
-  UIViewController *_controller;
+    __weak UIViewController * _Nullable _controller;
```
Comment (xuyanke) [inline Source/ObjectiveDropboxOfficial/Platform/ObjectiveDropboxOfficial_iOS/DBOAuthMobile-iOS.m:18]: nit: this file has 2-whitespace indent, maybe shorten this indent to align with other lines?

File: Source/ObjectiveDropboxOfficial/Platform/ObjectiveDropboxOfficial_macOS/DBOAuthDesktop-macOS.m [hunk 2]
```diff
@@ -11,7 +11,7 @@
 
 @implementation DBDesktopSharedApplication {
   NSWorkspace *_sharedWorkspace;
-  NSViewController *_controller;
+    __weak NSViewController * _Nullable _controller;
```
Comment (xuyanke) [inline Source/ObjectiveDropboxOfficial/Platform/ObjectiveDropboxOfficial_macOS/DBOAuthDesktop-macOS.m:14]: same nit here.

**Review by xuyanke — APPROVED** [review]
lgtm with minor comment about format.

**Comment by YufeiG** [comment]
Being able to edit code from the web browser is kinda nice

**Pull request merged**